### PR TITLE
DOC: add unqiue value counts example to groupyby guide

### PR DESCRIPTION
**Pull request by Abo7atm**
Co-authored-by: Abdulelah Almesfer <28743265+abdulelahsm@users.noreply.github.com>
Co-authoerd-by: Abdulellah Alnumay <33042538+Abo7atm@users.noreply.github.com>

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Adding an example to the groupby user guide. The example showcases the usage of ```nunique``` aggregate function. This is from the PyData Global sprint, from [this issue](https://github.com/MarcoGorelli/PyDataGlobal2020-sprint/issues/5). The example is taken from [StackOverflow](https://stackoverflow.com/questions/15411158/pandas-countdistinct-equivalent).

### CODE REVIEW

File: doc/source/user_guide/groupby.rst [hunk 1]
```diff
@@ -524,6 +524,13 @@ index are the group names and whose values are the sizes of each group.
 
    grouped.describe()
 
+Another aggregation example is to compute the number of unique values of a group. This is similar to the ``value_counts`` function, execpt that ``value_counts`` return the number of all values instead of the number of unique values.
```
Comment (MarcoGorelli) [inline doc/source/user_guide/groupby.rst:527]:
```suggestion
Another aggregation example is to compute the number of unique values of each group. This is similar to the ``value_counts`` function, except that it only counts unique values.
```

File: doc/source/user_guide/groupby.rst [hunk 2]
```diff
@@ -524,6 +524,13 @@ index are the group names and whose values are the sizes of each group.
...
+
+.. ipython:: python
+
+   ll = [['foo', 1], ['foo', 2], ['foo', 2], ['bar', 1], ['bar', 1]]
+   pd.DataFrame(ll, columns=['A', 'B']).groupby('A').nunique()
```
Comment (MarcoGorelli) [inline doc/source/user_guide/groupby.rst:532]:
```suggestion
   pd.DataFrame(ll, columns=["A", "B"]).groupby("A")["B"].nunique()
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @Abo7atm

For context, this is a common duplicate on StackOverflow, so I think it's good to include it in the docs.

Did you try building this page of the docs locally? If so, can you post a screenshot of this new section?

**Comment by Abo7atm** [comment]
Screen Shot 2020-11-15 at 18 40 51

File: doc/source/user_guide/groupby.rst [hunk 3]
```diff
@@ -524,6 +524,13 @@ index are the group names and whose values are the sizes of each group.
 
    grouped.describe()
 
+Another aggregation example is to compute the number of unique values of each group. This is similar to the ``value_counts`` function, except that it only counts unique values.
+
+.. ipython:: python
+
+   ll = [['foo', 1], ['foo', 2], ['foo', 2], ['bar', 1], ['bar', 1]]
+   pd.DataFrame(ll, columns=["A", "B"]).groupby("A")["B"].nunique()
```
Comment (jreback) [inline doc/source/user_guide/groupby.rst:532]: can you first show the input df, then do the groupby.nunique on the next line

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by Abo7atm** [comment]
@jreback done 👍

**Comment by jreback** [comment]
can you merge master @Abo7atm the precommit checks are failing

**Comment by Abo7atm** [comment]
@jreback merged master and ran precommit locally with `pre-commit run --files doc/source/user_guide/groupby.rst` and everything passed.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Abo7atm

**Pull request merged**